src/cryptography/x509.py

```diff
@@ -1594,3 +1596,138 @@ def sign(self, private_key, algorithm, backend):
         if self._subject_name is None:
             raise ValueError("A CertificateSigningRequest must have a subject")
         return backend.create_x509_csr(self, private_key, algorithm)
+
+
+class CertificateBuilder(object):
+    def __init__(self, version=None, issuer_name=None, subject_name=None,
+                 public_key=None, serial_number=None, not_valid_before=None,
+                 not_valid_after=None, extensions=[]):
+        """
+        Creates an empty X.509 certificate (version 1).
+        """
+        self._version = Version.v3
+        self._issuer_name = issuer_name
+        self._subject_name = subject_name
+        self._public_key = public_key
+        self._serial_number = serial_number
```

Comment:
Oops I see we *are* enforcing integer type in the setter, but not the non-negative or 160-bit restriction.